### PR TITLE
Add a genre filter enrichment

### PR DESCRIPTION
**Pull request by no-reply**
Removes values that do not match a given list of genre terms.

Also splits string enrichment shared examples to deal with string enrichments that don't want to ignore non-string values. The shared example which requires that is put into its own group so it can be combined as needed.

### CODE REVIEW

File: lib/krikri/enrichments/genre_filter.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module Krikri::Enrichments
+  ##
+  # Enrichment to remove non-genre fields from
+  #
+  #   StripHtml.new.enrich_value('Book') => 'Book'
+  #   StripHtml.new.enrich_value('book') => 'book'
```
Comment (markbreedlove) [inline lib/krikri/enrichments/genre_filter.rb]: It looks from lines 39-43 like this will result in "Book"  with a capital "B".

Comment (no-reply) [inline lib/krikri/enrichments/genre_filter.rb]: That's right... this comment doesn't belong, I think.

**Comment by markbreedlove** [comment]
Looks good to go, unless that comment needs to be changed.

**Comment by no-reply** [comment]
I'll change it.

**Comment by no-reply** [comment]
Comment fixed. I think this should stay two commits, so unless there's disagreement there, this is ready to :merge: from my standpoint.

**Pull request merged**